llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -367,6 +372,16 @@ class RISCVPassConfig : public TargetPassConfig {
       DAG->addMutation(createStoreClusterDAGMutation(
           DAG->TII, DAG->TRI, /*ReorderWhileClustering=*/true));
     }
+
+    const RISCVSubtarget &ST = C->MF->getSubtarget<RISCVSubtarget>();
+    if (!ST.getMacroFusions().empty()) {
+      DAG = DAG ? DAG : createGenericSchedLive(C);
+
+      if (ST.useLoadStorePairs()) {
```

Comment:
I think this duplicates code above?

Reply:
Yes, thanks, addressed in https://github.com/llvm/llvm-project/pull/121394